OpenDreamRuntime/Objects/Types/DreamObjectException.cs

```diff
@@ -1,17 +1,38 @@
 namespace OpenDreamRuntime.Objects.Types;
 
 public sealed class DreamObjectException(DreamObjectDefinition objectDefinition) : DreamObject(objectDefinition) {
-    public string Name = string.Empty;
+    public DreamValue Name = DreamValue.Null;
     public string Description = string.Empty;
     public string File = string.Empty;
     public int Line = 0;
 
     //TODO: Match the format of BYOND exceptions since SS13 does splittext and other things to extract data from exceptions
 
+    protected override void SetVar(string varName, DreamValue value) {
+        switch (varName) { //set the internal vars of the C# object, then pass through to base.setvar to handle DM side
+            case "name":
+                Name = value;
+                break;
+            case "desc":
+                if(value.TryGetValueAsString(out var stringDescription))
+                    Description = stringDescription;
+                break;
+            case "file":
+                if(value.TryGetValueAsString(out var stringFile))
+                    File = stringFile;
+                break;
+            case "line":
+                if(value.TryGetValueAsInteger(out var intLine))
+                    Line = intLine;
+                break;
+        }
+        base.SetVar(varName, value);
```

Comment:
`base.SetVar()` should only be called in the default case of the above switch. This is because it will add any variables it doesn't recognize to the object's list of modified variables. This is unnecessary when it will never be used.